core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -5369,6 +5493,16 @@ private void guardAgainstDotsInOrderByAdvice(QueryModel model) throws SqlExcepti
         }
     }
 
+    private void guardAgainstFromToWithKeyedSampleBy(boolean isFromTo) throws SqlException {
+        if (isFromTo) {
+            throw SqlException.$(0, "FROM-TO intervals are not supported for keyed sample by queries");
+        }
+    }
+
+    private boolean isFillStridePresent(QueryModel model) {
```

Comment:
nit: this method doesn't clarify things on top of `model.getFillStride() != null`, so maybe it's not worth having it?